### ISSUE TITLE
mmu port flush timeout

### ISSUE THREAD

**Issue by stigt** [platinasystems/go#39]
New code was adding this morning that flushes the mmu for Issue #1.  Sometimes it will crash when admin up/down an interface.  Eliot & Don are investigating.

Apr 19 15:07:11 invader24 kernel: ixgbe 0000:03:00.0 eth1: VF Reset msg received from vf 0
Apr 19 15:07:12 invader24 goes.vnetd[8382]: mmu port flush timeout true: goroutine 27 [running]:
Apr 19 15:07:12 invader24 goes.vnetd[8382]: runtime/debug.Stack(0x0, 0x0, 0x0)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /usr/local/go/src/runtime/debug/stack.go:24 +0x80
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/elib/loop.(*Loop).doEvent.func1(0xc820192000)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/elib/loop/event.go:128 +0xa2
Apr 19 15:07:12 invader24 goes.vnetd[8382]: panic(0xd9b0a0, 0xc8211b4950)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /usr/local/go/src/runtime/panic.go:426 +0x4e9
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a.(*Port).mmu_port_flush(0xc82036b628, 0xc82021e000)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a/mmu.go:391 +0x27f
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a.(*fe1a).swIfAdminUpDown(0xc82021e000, 0xc820192000, 0x44, 0x0, 0x0)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a/l3.go:62 +0x2ff
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a.(*fe1a).(github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a.swIfAdminUpDown)-fm(0xc820192000, 0x44, 0x0, 0x0)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/vnet/devices/ethernet/switch/fe1/internal/fe1a/l3.go:26 +0x49
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/vnet.(*swIf).SetAdminUp(0xc820666840, 0xc820192000, 0xc820178100, 0x0, 0x0)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/vnet/interface.go:225 +0xce
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/vnet.Si.SetAdminUp(0xc800000044, 0xc820192000, 0xffffff00, 0x0, 0x0)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/vnet/interface.go:235 +0x8a
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/vnet/unix.(*netlinkEvent).EventAction(0xc8208ee0c0)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/vnet/unix/netlink.go:330 +0x8fa
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/elib/loop.(*loopEvent).do(0xc8208ee120)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/elib/loop/event.go:118 +0x135
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/elib/loop.(*Loop).doEvent(0xc820192000, 0xc8208ee120)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/elib/loop/event.go:133 +0x48
Apr 19 15:07:12 invader24 goes.vnetd[8382]: github.com/platinasystems/go/elib/loop.(*Loop).eventHandler(0xc820192000, 0x7f5330fe3608, 0xc820192548)
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/elib/loop/event.go:140 +0x81
Apr 19 15:07:12 invader24 goes.vnetd[8382]: created by github.com/platinasystems/go/elib/loop.(*Loop).startHandler
Apr 19 15:07:12 invader24 goes.vnetd[8382]:         /home/stig/go/src/github.com/platinasystems/go/elib/loop/event.go:153 +0x120
Apr 19 15:07:12 invader24 goes.vnetd[8382]: done

**Comment by stigt** [comment]
The following script was used to test on Invader24 while 36Gb of traffic flowing through all interfaces.
```
#!/bin/bash

check_fib () {
  VAL=$(goes vnet show ip fib | grep eth-$1-1 | wc -l)
  echo $VAL
}

is_dead () {
  OUTPUT=$(goes vnet show ip fib 2>&1)
  if [[ "$OUTPUT" =~ "timeout" ]]
  then
     echo "vnet timeout"
     exit 1
  fi
}

bounce () {
  i=$1
  export IFACE=eth-$i-1
  is_dead
  echo -n "  FIB $IFACE before "
  B4=$(check_fib $i)
  is_dead
  echo $B4
  ifconfig $IFACE down
  is_dead
  echo -n "  FIB down "
  DOWN=$(check_fib $i)
  echo $DOWN
  if [ "$DOWN" -ne "0" ]
  then
     echo "FIB entries found for $IFACE = $DOWN"
     exit 1
  fi
  ifconfig $IFACE up
  /etc/network/if-up.d/static_arp
  #sleep 10
  is_dead
  echo -n "  FIB up "
  AFTR=$(check_fib $i)
  echo $AFTR
  if [ "$B4" -ne "$AFTR" ]
  then
     echo "  FIB missmatch $B4 $AFTR"
     exit 1
  fi
}

STOP=0
if [ $# -eq 1 ]
then
  STOP=$1
fi

LOOPS=1

while true
do
  echo Interation $LOOPS
  for i in `seq 1 32`
  do
    bounce $i
  done
  if [ "$LOOPS" -eq "$STOP" ]
  then
     echo Done - $LOOPS interations
     exit 0
  fi
  ((LOOPS++))
done
```

Example run.  Note took 5 minutes and 8 iterations through all 32 interfaces:

```
Interation 8
  FIB eth-1-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-2-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-3-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-4-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-5-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-6-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-7-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-8-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-9-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-10-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-11-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-12-1 before 18
  FIB down 0
  FIB up 18
  FIB eth-13-1 before 18
  FIB down vnet: /run/goes/socks/vnet: timeout
0
vnet timeout

real	5m36.784s
user	7m20.084s
sys	1m16.804s
```

Appears to happen most often on "ifconfig <intf> down".

**Comment by fszyang** [comment]
Closed.  Duplicate of issue #45
Fixed in fe1 version 4817d1d90a9e5610cd44a71d23f68c1b5726d3d0
Verified in goes version 98f3841
fe1: f823d6e1f564397570e68f4f1e2d43c02ccdc79c

Tested both fixed-100g and autoneg ports. No more panics.
